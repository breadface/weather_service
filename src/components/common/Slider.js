//@flow
import React from 'react';

type SliderType = {
  min: number,
  max: number,
  value: number,
  title: string,
  onChange: (value: number) => mixed
}

const Slider = ({onChange, min, max, value, title}: SliderType) =>  (
  <div className="grid-cell center slider-container">
    <div className="slider">
      <div className="title">{title}</div>
      <div>
        <input
          type="range"
          value={value}
          min={min} max={max}
          onChange={e => {
            onChange(e.target.value);
          }}
        />
      </div>
      <div className="slider-value">
        <div>{value}</div>
      </div>
    </div>
  </div>
);

export default Slider
