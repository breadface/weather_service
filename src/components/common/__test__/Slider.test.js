import React from 'react';
import renderer from 'react-test-renderer'
import store from  '../../../store'
import Slider from '../Slider';

test('Renders snapshot for Weather Slider page', () => {
  const tree = renderer.create(
    <Slider />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})
