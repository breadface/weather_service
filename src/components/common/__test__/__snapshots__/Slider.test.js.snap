// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders snapshot for Weather Slider page 1`] = `
<div
  className="grid-cell center slider-container"
>
  <div
    className="slider"
  >
    <div
      className="title"
    />
    <div>
      <input
        max={undefined}
        min={undefined}
        onChange={[Function]}
        type="range"
        value={undefined}
      />
    </div>
    <div
      className="slider-value"
    >
      <div />
    </div>
  </div>
</div>
`;
